code/modules/reagents/chemistry/reagents.dm

```diff
@@ -97,6 +97,15 @@
 /datum/reagent/proc/on_mob_death(mob/living/M)	//use this to have chems have a "death-triggered" effect
 	return
 
+/datum/reagent/proc/fire_flash_log(datum/reagents/holder) //Call this whenever a chemical makes a fire flash
```

Comment:
Although it probably works this way, passing the holder arg seems unnecessary. The `/reagent`, or `/chemical_reaction` can already access their own `holder`s.

Reply:
Removed, kept on the recipe one as it is needed.